### ISSUE TITLE
`prefer-top-level-await`: Do not report cases where the promise is assigned to a variable

### ISSUE THREAD

**Issue by jhnns** [sindresorhus/eslint-plugin-unicorn#2516]
### Description

Hello 👋, thank you for this great plugin :)

I have a case where I create a `Promise` on the top-level and assign it to a variable. Later, in an asynchronous function, I await the promise.

The motivation behind this is as follows: I don't want to use top-level await because I don't want to slow down the evaluation of the module scope. The result of the Promise must only be available at the time when the asynchronous function is executed.

Would it therefore be possible not to report cases in which the Promise is assigned to a variable?

### Fail

```js
run().catch(error => {
	console.error(error);
	process.exit(1);
});
```

### Pass

```js
const resultOfRun = run();
```

### Additional Info

I know that the rule might be "too relaxed" in that case, but I still think that my use case is valid and should not reported as an anti-pattern :)

**Comment by sindresorhus** [comment]
I agree that this is a useful pattern and should not be reported on.

**Comment by jhnns** [comment]
Would you accept a PR for it? :)

**Comment by sindresorhus** [comment]
Yes
